### ISSUE TITLE
A linter for file names

### ISSUE THREAD

**Issue by PEZ** [clj-kondo/clj-kondo#842]
I can't count the times someone has committed a change where they've included a new Clojure source file, but used dashes instead of underscores in the file names, causing the build to fail.

How about if `clj-kondo` helped to spot this mistake?

I'd be happy to file a PR, if it seems like something that would fit the project.

**Comment by borkdude** [comment]
Seems good to me.

**Comment by borkdude** [comment]
Two things to think about:
1) Dashes are allowed in script files like `foo-bar.clj`, just not when used in a classpath (which is arguably most of the time). How should we let users communicate this to clj-kondo? When writing a script, what can users do to prevent the warning? 
2) Where do we display the warning, on the first line of the file?

**Comment by PEZ** [comment]
So I'll start with the one I think is easier: 2. I ran into this problem a few days ago. A babashka script I had was working nicely in the REPL, but then run from the command prompt it failed loading a require. Because I had named the required file with dashes. It would have made sense to me to have a warning displayed as a squiggly on the namespace symbol in the `ns` form.

As for 1. In my case the required namespace was also a script. So my initial idea with not warning if there was a shebang there would not have worked... One solution might be to warn only if there is an attempt to require the namespace. It's a bit ”better late than never”. 😄
